v21.1/use-cloud-storage-for-bulk-operations.md

```diff
@@ -96,75 +96,141 @@ CockroachDB also provides client-side encryption of backup data, for more inform
+As an example:
 
-    ~~~shell
-    aws ec2 associate-iam-instance-profile --iam-instance-profile Name={example profile} --region={us-east-2} --instance-id {instance example}
-    ~~~
+{% include_cached copy-clipboard.html %}
+~~~sql
+BACKUP DATABASE <database> INTO 's3://{bucket name}/{path in bucket}/?AWS_ACCESS_KEY_ID={access key ID}&AWS_SECRET_ACCESS_KEY={secret access key}';
+~~~
+
+### Implicit Authentication
 
-### Google Cloud Storage
+If the `AUTH` parameter is `implicit`, the access keys can be omitted and [the credentials will be loaded from the environment](https://docs.aws.amazon.com/sdk-for-go/api/aws/session/), i.e. the machines running the backup.
 
-If the `AUTH` parameter is set to:
+~~~sql
+BACKUP DATABASE <database> INTO 's3://{bucket name}/{path}?AUTH=implicit';
+~~~
+
+You [can associate an EC2 instance with an IAM role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) to provide implicit access to S3 storage within the IAM role's policy. In the following command, the `instance example` EC2 instance is [associated](https://docs.aws.amazon.com/cli/latest/reference/ec2/associate-iam-instance-profile.html) with the `example profile` instance profile, giving the EC2 instance implicit access to any `example profile` S3 buckets.
 
-  - `default`: GCS connections only use the key provided in the `cloudstorage.gs.default.key` [cluster setting](cluster-settings.html), and will error if not present.
-  - `specified`: Pass the JSON object for authentication to the `CREDENTIALS` parameter. The JSON key object needs to be base64-encoded (using the standard encoding in [RFC 4648](https://tools.ietf.org/html/rfc4648)).
+{% include_cached copy-clipboard.html %}
+~~~shell
+aws ec2 associate-iam-instance-profile --iam-instance-profile Name={example profile} --region={us-east-2} --instance-id {instance example}
+~~~
 
-    To access the storage bucket with specified credentials, it's necessary to [create a service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) and add the service account address to the permissions on the specific storage bucket. [The JSON object for authentication](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-console) can be downloaded, encoded, and then passed to the `CREDENTIALS` parameter.
+</section>
 
-    ~~~sql
-    BACKUP DATABASE <database> INTO 'gs://{bucket name}/{path}?AUTH=specified&CREDENTIALS={encoded key}';
+<section class="filter-content" markdown="1" data-scope="gcs">
+
+The `AUTH` parameter passed to the file URL must be set to either `specified` or `implicit`. The following sections describe how to set up each authentication method.
+
+In v21.1 and earlier, Google Cloud Storage connections can use `default`. This means that the connection will only use the key provided in the `cloudstorage.gs.default.key` [cluster setting](cluster-settings.html), and will error if not present.
```

Comment:
Is it worth reiterating that in <21.1 if no parameter was specified then we would default to `DEFAULT`, but 21.2 onwards we default to `SPECIFIED`?

Reply:
That's a good point Aditya, I hadn't specifically mentioned that `default` was the default in this paragraph. I have added/moved this around. And emphasized the warning on the v21.1 page that notes that `specified` is the default in v21.2+.